app/services/foreman_templates/template_importer.rb

```diff
@@ -163,5 +155,13 @@ def purge!
         template.destroy
       end
     end # :purge
+
+    private
+
+    def assign_attributes(args = {})
+      %w(verbose associate prefix dirname filter repo negate branch).each do |attribute|
+        instance_variable_set("@#{attribute}", args[attribute.to_sym] || Setting["templates_#{attribute}".to_sym])
```

Comment:
Some tests would be good. I think the existing tests will all hit the left-hand-side of this `||`, so some tests using `Setting` and some using both (showing that override works) would be great.